src/web/rustdoc.rs

```diff
@@ -301,21 +291,60 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     let latest_version = crate_details.latest_version().to_owned();
     let is_latest_version = latest_version == version;
 
-    let path = if !is_latest_version {
-        req_path[2] = &latest_version;
-        path_for_version(&req_path, &crate_details.target_name, &conn)
+    let path_in_latest = if !is_latest_version {
+        let mut latest_path = req_path.clone();
+        latest_path[2] = &latest_version;
+        path_for_version(&latest_path, &crate_details.target_name, &conn)
     } else {
         Default::default()
     };
 
+    // for each target the documentation is built for, check if the requested file also exists in
+    // that target and generate a list of targets + paths to render.
+    if let Some(Json::Array(doc_targets)) = &crate_details.doc_targets {
+        content.platforms = doc_targets
+            .iter()
+            .filter_map(|target| {
+                if let Json::String(target) = target {
+                    // rustdoc/:crate/:version[/:current_target]/:path*
+                    let mut platform_path = req_path.clone();
+                    // rustdoc/:crate/:version/:target/:path*
+                    if platform_path[3] == crate_details.target_name {
+                        platform_path.insert(3, target);
+                    } else {
+                        platform_path[3] = target;
+                    }
+                    if target == &crate_details.metadata.default_target {
+                        // rustdoc/:crate/:version/:path*
+                        platform_path.remove(3);
+                    }
+                    if File::from_path(&conn, &platform_path.join("/")).is_none() {
```

Comment:
```suggestion
                    // if the current page is not present on the other platform, link to the index page for the crate instead
                    if File::from_path(&conn, &platform_path.join("/")).is_none() {
```

Reply:
Is there a reason you didn't reuse `path_for_version` for this? That would also handle searching for the page if it doesn't exist on the other platform.

That function should probably be renamed to `link_or_search` or similar, now that I think about it...

Reply:
I doubt that searching would be that useful, and IIRC it didn't fit this case exactly. But I think now that I understand how this is all working much better I might be able to refactor that to fit both usecases.

Reply:
Hmm I feel like searching would still be useful. For example [winapi::winrt::winstring::HSTRING_UserFree64](https://docs.rs/winapi/0.3.8/winapi/winrt/winstring/fn.HSTRING_UserFree64.html) is [only present on `x86_64-pc-windows-msvc`](https://docs.rs/winapi/0.3.8/i686-pc-windows-msvc/winapi/winrt/winstring/fn.HSTRING_UserFree64.html), but if you search for it on `i686-pc-windows-msvc` [a similar function comes up](https://docs.rs/winapi/0.3.8/i686-pc-windows-msvc/winapi/winrt/winstring/fn.HSTRING_UserFree.html).

Reply:
Ah, yeah, 32 vs 64 bit differences make sense; and I guess crates with types like `OsExt` might have them in platform specific locations but providing equivalent functionality.